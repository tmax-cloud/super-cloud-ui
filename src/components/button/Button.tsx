import * as React from 'react';
import { Button as MuiButton, styled, SxProps, Theme } from '@mui/material';
import { CommonProps } from '@mui/material/OverridableComponent';
import ThemeWrapper from '../../themes/ThemeWrapper';

const StyledButton = styled(MuiButton)(({ theme }) => ({
  textTransform: 'none',
  backgroundColor: theme.palette.button.primaryBg,
  '&:hover': {
    backgroundColor: theme.palette.button.primaryHoverBg,
  },
  '&.Mui-disabled': {
    color: theme.palette.button.disabled,
    backgroundColor: theme.palette.button.disabledBg,
    cursor: 'no-drop',
    pointerEvents: 'auto',
  },
}));

export const Button = (props: ButtonProps) => {
  const { children, ...rest } = props;
  return (
    <ThemeWrapper>
      <StyledButton variant="contained" {...rest}>
        {children}
      </StyledButton>
    </ThemeWrapper>
  );
};

interface BaseProps {
  /**
   * The content of the component.
   */
  children?: React.ReactNode;
  /**
   * If `true`, the component is disabled.
   * @default false
   */
  disabled?: boolean;
  /**
   * The URL to link to when the button is clicked.
   * If defined, an `a` element will be used as the root node.
   */
  href?: string;
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx?: SxProps<Theme>;
}

export type ButtonProps = BaseProps & CommonProps & React.DOMAttributes<HTMLButtonElement>;

Button.defaultProps = {
  disabled: false,
};
