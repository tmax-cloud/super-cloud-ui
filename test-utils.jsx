// import React from 'react';
// import { render } from '@testing-library/react';
// 추후에 redux 추가하게 되면 redux Provider로 감싸주는 custom render 메서드 추가 해야됨.

export * from '@testing-library/react';
