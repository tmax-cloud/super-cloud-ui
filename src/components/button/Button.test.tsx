import * as React from 'react';
import { render } from '../../../test-utils';
import Button, { ButtonType } from './Button';

describe('Button', () => {
  Object.values(ButtonType).forEach((type) => {
    test(`${type} button`, () => {
      const view = render(<Button type={type}>{type} Button</Button>);
      expect(view.container).toMatchSnapshot();
    });
  });
  test('disabled button', () => {
    const { container } = render(<Button disabled>Disabled Button</Button>);
    expect(container).toMatchSnapshot();
  });
});
