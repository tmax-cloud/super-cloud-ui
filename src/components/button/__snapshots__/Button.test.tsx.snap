// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-7ifs1y-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    Button
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
